### PR TITLE
Contrib: Glue code to use rofi as a menu replacement/alternative

### PR DESCRIPTION
**Pull request by olejorgenb**
[Rofi](https://github.com/DaveDavenport/rofi)

Adds `rofi.menu` - a `mod_menu.menu` compatible (mostly at least) replacement/alternative. 

An advantage over notion's menus is that it support fuzzier matching. (each word in the query is matched independently against the menu content making it order-independent for instance. The menu is also interactively filtered down.

Rofi support a built-in mode for switching windows which works with notion (when net_client_list is loaded), but the windows isn't listed in MRU order, and the rofi's built in workspace menu doesn't work IIRC.

Atm only regular menus is supported, but I don't think it would be very hard to support context menus too.

The mechanism used for passing data is a bit messy (suggestions welcome) but it works. 

Coauthored with @hedning 

2017-04-11-063312_1366x768_scrot

### CODE REVIEW

File: contrib/scripts/notion_rofi.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env zsh
+#
+# Example:
+# rofi -show notion -modi "notion:'./notion_rofi.sh focuslist'" -scroll-method 1
+
+function escape {
+    sed "s/'/\\\'/g"
```
Comment (raboof) [inline contrib/scripts/notion_rofi.sh]: Shouldn't existing `\` characters be somehow escaped, too?

**Comment by raboof** [comment]
I'm not sure I completely understand how this works (and whether it makes sense to integrate it into Notion at all - couldn't it be completely separate?), but as it's in contrib I think it's fair game.

Thanks for the contribution!

File: contrib/scripts/notion_rofi.sh (hunk 1 above)
Comment (olejorgenb) [inline contrib/scripts/notion_rofi.sh]: Yeah, you're right :)

It's not possible to activate a menu entry containing `\'`

**Comment by olejorgenb** [comment]
Yeah, it's a bit messy, but I think it's kinda in the nature of how rofi and notionflux work.

**Comment by olejorgenb** [comment]
> and whether it makes sense to integrate it into Notion at all - couldn't it be completely separate?

Some of it can be completely separate. "Go to window" is a built-in function in rofi and works when `net_client_list` is loaded into notion. But for stuff like workspaces, etc. I don't see how it's possible to not involve notion. (the list of window titles in the built-in rofi mode isn't MRU sorted either)

**Pull request merged**